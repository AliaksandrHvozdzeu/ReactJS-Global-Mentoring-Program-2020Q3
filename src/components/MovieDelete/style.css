.delete-movie {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 2;
}


.delete-modal {
    height: 10rem;
}

.modal-title {
    display: block;
    color: var(--white);
    text-transform: uppercase;
    font-weight: lighter;
    font-size: 2.5rem;
    letter-spacing: 0.1rem;
    margin-bottom: 1.1rem;
}

.delete-movie-description {
    font-size: 1.7rem;
    font-weight: 200;
    margin: 1rem 0;
    color: var(--white);
}

.submit-delete {
    font-size: 1.2rem;
    font-weight: 200;
    height: 3rem;
    padding: 0.5rem 4rem;
    color: var(--white);
    background-color: var(--red);
    text-transform: uppercase;
    border-radius: 4px;
    outline: none;
    letter-spacing: 0.1rem;
    position: absolute;
    right: 30px;
}

.submit-delete:hover {
    background-color: var(--red-lighter);
}

.submit-delete:active {
    box-shadow: inset 0 0 0 1px #000000;
    background-color: var(--red-darker);
}
