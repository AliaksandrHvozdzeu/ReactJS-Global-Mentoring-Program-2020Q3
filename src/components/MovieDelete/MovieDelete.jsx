import React from 'react';
import PropTypes from 'prop-types';
import CloseButton from '../CloseButton';
import './style.css';

export default function MovieDelete({ modalTitle, closeAction, deleteAction }) {

  return (
    <>
      <CloseButton closeAction={closeAction}/>
      <section className="delete-modal">
        <p className="modal-title">{modalTitle}</p>
        <p className="delete-movie-description">
          Are you sure you want to delete this movie?
        </p>
        <button type="button" onClick={deleteAction} className="submit-delete">Confirm</button>
      </section>
    </>
  );

}

MovieDelete.propTypes = {
  modalTitle: PropTypes.string.isRequired,
  closeAction: PropTypes.func.isRequired,
  deleteAction: PropTypes.func.isRequired,
};
