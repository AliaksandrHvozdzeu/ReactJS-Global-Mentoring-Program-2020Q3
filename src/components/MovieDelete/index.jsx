import React from 'react';
import PropTypes from 'prop-types';
import './style.css';
import CloseButton from '../CloseButton';

const MovieDelete = ({ modalTitle, closeAction, deleteAction }) => (
  <>
    <CloseButton closeAction={closeAction}/>
    <section className="delete-modal">
      <p className="modal-title">{modalTitle}</p>
      <p className="delete-movie-description">
        Are you sure you want to delete this movie?
      </p>
      <button onClick={deleteAction} className="submit-delete">Confirm</button>
    </section>
  </>
);

MovieDelete.propTypes = {
  modalTitle: PropTypes.string.isRequired,
  closeAction: PropTypes.func.isRequired,
  deleteAction: PropTypes.func.isRequired
};

export default MovieDelete;
