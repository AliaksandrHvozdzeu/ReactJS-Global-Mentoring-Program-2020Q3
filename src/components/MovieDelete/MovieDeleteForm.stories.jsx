import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import MovieDeleteForm from './js/MovieDeleteForm';
import './css/MovieDelete.css';
import '../../App.css';
import CloseButton from "../CloseButton";

export default {
  title: 'MovieDeleteForm',
  decorators: [withKnobs],
};

export const Default = () => (
  <>
    <CloseButton closeAction={() => alert('closeAction')}/>
    <section className="delete-modal">
      <p className="modal-title">MovieDeleteForm</p>
      <p className="delete-movie-description">
        Are you sure you want to delete this movie?
      </p>
      <button type="button"
              onClick={() => alert('onMovieDeletion')}
              className="submit-delete">
        Confirm
      </button>
    </section>
  </>
);
