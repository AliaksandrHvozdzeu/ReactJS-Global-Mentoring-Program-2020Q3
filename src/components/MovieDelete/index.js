import MovieDelete from './js/MovieDelete';

export default MovieDelete;
