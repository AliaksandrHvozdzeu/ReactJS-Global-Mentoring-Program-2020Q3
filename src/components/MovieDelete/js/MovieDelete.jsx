import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import '../css/MovieDelete.css';
import movieActions from '../../../store/actions/movie';
import MovieDeleteForm from './MovieDeleteForm';

const MovieDelete = ({ modalTitle, closeAction, onMovieDeletion, details }) => {

  return (
    <MovieDeleteForm modalTitle={modalTitle}
                     closeAction={closeAction}
                     onMovieDeletion={onMovieDeletion}
                     details={details}/>
  );

};

MovieDelete.propTypes = {
  details: PropTypes.shape({
    title: PropTypes.string,
    vote_average: PropTypes.string,
    genres: PropTypes.arrayOf(PropTypes.string),
    release_date: PropTypes.string,
    runtime: PropTypes.number,
    overview: PropTypes.string,
  }).isRequired,
  modalTitle: PropTypes.string.isRequired,
  closeAction: PropTypes.func.isRequired,
  onMovieDeletion: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  movies: state.movies.details,
});

const mapDispatchToProps = (dispatch) => ({
  onMovieDeletion: (details) => dispatch(movieActions.deleteMovie(details)),
});

export default connect(mapStateToProps, mapDispatchToProps)(MovieDelete);
