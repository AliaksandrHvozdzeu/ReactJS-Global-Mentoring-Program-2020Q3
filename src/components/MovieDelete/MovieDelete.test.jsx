import React from "react";
import rerender from "react-test-renderer";
import { jest } from "@jest/globals";
import MovieDelete from "./js/MovieDelete";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

describe("when MovieDelete", () => {

  test("then snapshot created", () => {

    const modalTitle = "DELETE";
    const closeAction = jest.fn();
    const onMovieDeletion = jest.fn();
    const details = {
      "id": 337167,
      "title": "Fifty Shades Freed",
      "tagline": "Don't miss the climax",
      "vote_average": 6.1,
      "vote_count": 1195,
      "release_date": "2018-02-07",
      "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
      "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
      "budget": 55000000,
      "revenue": 136906000,
      "genres": [
        "Drama",
        "Romance"
      ],
      "runtime": 106
    };
    const initialStore = {
      movies: {
        "genres": ["Comedy", "Crime", "Documentary", "Horror"]
      }
    };
    const store = configureStore([])(initialStore);
    store.dispatch = jest.fn();
    const component = rerender.create(<Provider store={store}><MovieDelete modalTitle={modalTitle}
                                                                           closeAction={closeAction}
                                                                           onMovieDeletion={onMovieDeletion}
                                                                           details={details} /> </Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
