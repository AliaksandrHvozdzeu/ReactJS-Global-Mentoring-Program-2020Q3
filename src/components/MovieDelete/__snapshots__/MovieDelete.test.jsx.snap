// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when MovieDelete then snapshot created 1`] = `
Array [
  <button
    aria-label="close"
    className="close-button"
    onClick={[MockFunction]}
    type="button"
  />,
  <section
    className="delete-modal"
  >
    <p
      className="modal-title"
    >
      DELETE
    </p>
    <p
      className="delete-movie-description"
    >
      Are you sure you want to delete this movie?
    </p>
    <button
      className="submit-delete"
      onClick={[Function]}
      type="button"
    >
      Confirm
    </button>
  </section>,
  " ",
]
`;
